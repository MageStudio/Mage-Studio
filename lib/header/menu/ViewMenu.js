import React, { Component } from 'react';
import { Dropdown, Menu, Divider } from 'antd';

const { SubMenu, Item } = Menu;

const getSidebarMenuContent = () => (
    <SubMenu title="Sidebars">
        <Item
            className="option"
            key="1">option 1</Item>
        <Item
            className="option"
            key="2">option 2</Item>
    </SubMenu>
);

const getMenuContent = () => (
    <Menu>
        <Item
            className="option"
            key="save_scene">
            <a
                href='/code'
                target='_blank'>
                Code Editor
            </a>
            </Item>
        <Item
            className="option"
            key="save_file">
            <a
                href='/'
                target='_blank'>
                Scene Editor
            </a>
        </Item>
        <Divider/>
        <Item
            className="option"
            key="toggle_fullscreen">
            Toggle Full Screen
        </Item>
        { getSidebarMenuContent() }
    </Menu>
)

export const ViewMenu = () => (
    <Dropdown overlay={getMenuContent()}>
        <span className="main-menu-item">
            View
        </span>
    </Dropdown>
);




export default ViewMenu;
