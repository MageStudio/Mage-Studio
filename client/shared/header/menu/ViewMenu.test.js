import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import ViewMenu from './ViewMenu';

describe('ViewMenu', () => {

    it('should render fine', () => {
        const component = shallow(<ViewMenu />);
        expect(toJSON(component)).toMatchSnapshot();
    });
})
