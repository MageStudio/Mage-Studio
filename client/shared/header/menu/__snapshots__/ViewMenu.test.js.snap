// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewMenu should render fine 1`] = `
<Dropdown
  mouseEnterDelay={0.15}
  mouseLeaveDelay={0.1}
  overlay={
    <Menu>
      <MenuItem
        className="option"
      >
        <a
          href="/code"
          target="_blank"
        >
          Code Editor
        </a>
      </MenuItem>
      <MenuItem
        className="option"
      >
        <a
          href="/"
          target="_blank"
        >
          Scene Editor
        </a>
      </MenuItem>
      <Divider />
      <MenuItem
        className="option"
      >
        Toggle Full Screen
      </MenuItem>
      <SubMenu
        title="Sidebars"
      >
        <MenuItem
          className="option"
        >
          option 1
        </MenuItem>
        <MenuItem
          className="option"
        >
          option 2
        </MenuItem>
      </SubMenu>
    </Menu>
  }
>
  <span
    className="main-menu-item"
  >
    View
  </span>
</Dropdown>
`;
